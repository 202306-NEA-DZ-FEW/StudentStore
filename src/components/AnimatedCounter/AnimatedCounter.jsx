import React, { useEffect, useState } from "react";

const AnimatedCounter = ({ targetCount }) => {
    const [count, setCount] = useState(0);

    useEffect(() => {
        if (count < targetCount) {
            const increment = Math.ceil(targetCount / 200); // Adjst the increment value as the sped u want to show
            const timeout = setTimeout(() => {
                setCount((prevCount) => prevCount + increment);
            }, 30); // Adjust the timeout duration as needed
            return () => clearTimeout(timeout);
        }
    }, [count, targetCount]);

    return (
        <div className='counter'>
            <div className='number font-bold text-5xl text-center transition duration-500 transform animate-fade-in'>
                {count}
            </div>
        </div>
    );
};

export default AnimatedCounter;
