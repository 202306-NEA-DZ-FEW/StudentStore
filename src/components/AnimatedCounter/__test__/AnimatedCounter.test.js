import renderer from "react-test-renderer";
import AnimatedCounter from "../AnimatedCounter";

it("renders correctly", () => {
    const tree = renderer.create(<AnimatedCounter />).toJSON();
    expect(tree).toMatchSnapshot();
});
