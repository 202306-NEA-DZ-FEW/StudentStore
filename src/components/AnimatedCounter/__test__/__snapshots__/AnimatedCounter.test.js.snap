// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="counter"
>
  <div
    className="number font-bold text-5xl text-center transition duration-500 transform animate-fade-in"
  >
    0
  </div>
</div>
`;
